debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/MySqlConnectorConfig.java

```diff
@@ -903,6 +1010,57 @@ private static int validateServerNameIsDifferentFromHistoryTopicName(Configurati
         return 0;
     }
 
+    private static int validateSnapshotLockingMode(Configuration config, Field field, ValidationOutput problems) {
+        // Grab configured values
+        final String lockingModeValueStr = config.getString(MySqlConnectorConfig.SNAPSHOT_LOCKING_MODE);
+        final String snapshotModeValueStr = config.getString(MySqlConnectorConfig.SNAPSHOT_MODE);
+        final boolean minimalLocksEnabled = config.getBoolean(MySqlConnectorConfig.SNAPSHOT_MINIMAL_LOCKING);
+
+        // Convert into appropriate enum values
+        final SnapshotLockingMode lockingModeValue = SnapshotLockingMode.parse(lockingModeValueStr);
+        final SnapshotMode snapshotModeValue = SnapshotMode.parse(snapshotModeValueStr);
+
+        // Sanity check, validate the configured value is a valid option.
+        if (lockingModeValue == null) {
+            problems.accept(SNAPSHOT_LOCKING_MODE, lockingModeValue, "Must be a valid snapshot.locking.mode value");
+            return 1;
+        }
+
+        // Validate that if deprecated snapshot.minimal.locks is set to 'true', locking mode must be 'minimal'
+        final Set<SnapshotLockingMode> validMinimalLockingModes = new HashSet<SnapshotLockingMode>() {
+            {
+                add(SnapshotLockingMode.MINIMAL);
+                add(SnapshotLockingMode.NONE);
+            }
+        };
+
+        // If minimal locks is enabled, we must have locking mode set to minimal or none.
+        if (minimalLocksEnabled && !validMinimalLockingModes.contains(lockingModeValue)) {
```

Comment:
Ah, that's interesting, so it seems you try to ensure a consistent configuration via the old and the new option. I was rather assuming that the new config would "win", and we'd log a warning if the old config has been given, too, and differs from the new one. WDYT?

Reply:
So the only issue I see with your approach is we set default values.  So imagine the following scenario.

- User sets the OLD minimal locking setting to false, meaning they want all the locking.
- User does NOT define the NEW snapshot locking mode setting, which has a default value equal to minimal = true.

In that scenario if we allow the new config to win, we've just changed the behavior on them even tho they've explicitly told it to do otherwise.  Does that make sense?

Reply:
Agreed, if an explicit value has been given for the old option but not for the new one, then that old value should be applied and not the default for the new one. I think you should be able to detect this via `Configuration#getKeys()`.